core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -1354,6 +1395,22 @@ class KafkaController(val config: KafkaConfig, zkUtils: ZkUtils, time: Time, met
 
   }
 
+  case class LogDirEventNotification(sequenceNumbers: Seq[String]) extends ControllerEvent {
+
+    def state = ControllerState.LogDirChange
+
+    override def process(): Unit = {
```

Comment:
Good point. I have added the following methods in `LogDirUtils` and invoked this at the beginning of `onControllerFailover()`.

```
def deleteLogDirEvents(zkUtils: ZkUtils) {
  val sequenceNumbers = zkUtils.getChildrenParentMayNotExist(ZkUtils.LogDirEventNotificationPath).toSet
  sequenceNumbers.map(x => zkUtils.deletePath(ZkUtils.LogDirEventNotificationPath + "/" + x))
}
```